core/base/topoMap/TopoMap.h

```diff
@@ -0,0 +1,1045 @@
+                           std::vector<size_t> &idsInHull) const;
+  };
+
+  // Sketch of the algorithm:
+  // 1. Sort the edges
+  // 2. For each edge uv with cost c_uv
+  //      a. Get connected components comp(u), comp(v)
+  //
+  //      b. Compute convex hull of comp(u) and of (comp(v))
+  //
+  //      c. Find in hull(u) (resp. hull(v)) the point pU (resp. pV) in the
+  //      convex hull which is closest to u (resp. v) in high dimension, u if
+  //      possible.
+  //
+  //      d. Translate comp(u) such that pU lies on the line of the bissector of
+  //      angleV, at distance c_uv of pV, and on the opposite direction of
+  //      hull(v). angleU (resp. angleV) is the angle of hull(u) (resp. hull(v))
+  //      at vertex pU (resp. pV).
+  //
+  //      e. Rotate comp(u) such that the bissector of angleU prolongates the
+  //      one of angleU, and such that pU and pV are closest than any pair of
+  //      points in comp(u),comp(v): the two components are facing each other at
+  //      pU and pV.
+  //
+  //
+  //      f. Try several rotations of comp(u) and comp(v) such that pU and pV
+  //      still are the closest pair of vertices from the two components. We
+  //      keep the angle of rotation which minimizes the difference between the
+  //      high dimension distance matrix and the new distance matrix between the
+  //      two components.
+  //
+  //      g. comp(u) and comp(v) were merged, we now consider them as a single
+  //      component.
+
+  template <typename T>
+  int ttk::TopoMap::execute(T *outputCoords,
+                            int *insertionTime,
+                            const std::vector<T> &inputMatrix,
+                            bool isDistMat,
+                            size_t n) {
+    ttk::Timer timer;
+
+#ifdef TTK_ENABLE_QHULL
+#ifndef Qhull_FOUND
+    this->printWrn(
+      "Qhull was enabled but it is not installed or was not found.");
+    this->printWrn("Defaulting to Boost support instead.");
+#endif
+#endif
+
+#ifndef Qhull_FOUND
+    this->printMsg("Using Boost for convex hulls.");
+    this->printWrn("Bugs have been reported in Boost's implementation.");
+    this->printWrn("Consider enabling Qhull instead.");
+#endif
+
+    if(AngularSampleNb < 2) {
+      this->printWrn("The number of angular samples set is less than 2.");
+      this->printWrn("Setting it to 2 (minimum possible value).");
+      this->AngularSampleNb = 2;
+    }
+    // We first set the minimum value we consider as null. We multiply by 100
+    // the smallest significative value because we use sqrt/cos/sin so we take
+    // precautionary measures.
+    if(std::is_same<T, double>::value) {
+      Epsilon = ttk::Geometry::pow(10.0, -DBL_DIG + 2);
+    } else if(std::is_same<T, float>::value) {
+      Epsilon = ttk::Geometry::pow(10.0, -FLT_DIG + 2);
+    } else {
+      printErr("Error, template type must be either double or float.\n");
+      return 1;
+    }
+#ifndef TTK_KAMIKAZE
```

Comment:
I'd remove this `ifndef` (and always execute the associated code block).
(plus, the actual flag is `TTK_ENABLE_KAMIKAZE`)